engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java

```diff
@@ -875,7 +875,8 @@ public static String getSocketConnectString(InetSocketAddress socketAddress)
   public static void addAttributeToArgs(Attribute<String> attribute, Context context, List<CharSequence> vargs)
   {
     if (context.getValue(attribute) != null) {
-      vargs.add(String.format("-D%s=$'%s'", attribute.getLongName(), context.getValue(attribute).replaceAll("['\"$\\\\]", "\\\\$0")));
+      vargs.add(String.format("-D%s=$'%s'", attribute.getLongName(), context.getValue(attribute)
+          .replaceAll("[\\\\]", "\\\\\\\\\\\\\\\\").replaceAll("['\"$]", "\\\\$0")));
```

Comment:
Replacing "\" does not require regular expression, use replace("\\", "\\\\\\\\") or replace("\\", "\\x5C") instead of replaceAll().